Framework/Algorithms/src/MonteCarloAbsorption.cpp

```diff
@@ -101,11 +101,18 @@ void MonteCarloAbsorption::init() {
   declareProperty(std::make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
                                                         Direction::Output),
                   "The name to use for the output workspace.");
+
+  declareProperty("SimulateTracksForEachWavelength", false,
+                  "Resimulate tracks for each wavelength point.");
   auto positiveInt = boost::make_shared<Kernel::BoundedValidator<int>>();
   positiveInt->setLower(1);
   declareProperty("NumberOfWavelengthPoints", EMPTY_INT(), positiveInt,
                   "The number of wavelength points for which a simulation is "
                   "attempted (default: all points)");
+  setPropertySettings("NumberOfWavelengthPoints",
+                      std::make_unique<EnabledWhenProperty>(
```

Comment:
I don't understand why `NumberOfWavelengthPoints` is enabled only when the above is true. Can you explain this?

Reply:
NumberOfWavelengthPoints was originally introduced to provide better performance on large calculations where the time spent generating the track geometry on each iteration of the MC simulation was large. The idea was that you'd just do the calculation on a subset of the wavelength bins and then interpolate between them.

As a result of this PR, NumberOfWavelengthPoints is not needed in most cases because the track geometry is calculated once and reused for each wavelength. As a result of this, it's possible to fully calculate the absorption factor on each bin in the input workspace without needing to subselect a few points and interpolate between them.
You could ask whether it's worth removing NumberOfWavelengthPoints entirely - I've left it in because it's possible there could be additional effects added to the MC simulation in future that may be wavelength dependent.